### PR TITLE
Allow setting Allauth provider secrets from host system

### PR DESCRIPTION
**Pull request by agjohnson**
This requires

- https://github.com/readthedocs/common/pull/209

It is modeled after the Stripe setting pattern. There is a stronger pattern here at some point, this is a bare minimum for now to unblock us from not having these settings on our local environments.

You can use `direnv` to automatically export (and encrypt if you are so inclined) these secrets for your local envs. The env vars pass through Docker with common/#209 above.

### CODE REVIEW

**Review by stsewd — APPROVED** [review]
We should have these documented in our dev docs.

File: readthedocs/settings/docker_compose.py [hunk 1]
```diff
@@ -208,6 +208,22 @@ def DATABASES(self):  # noqa
     STRIPE_TEST_SECRET_KEY = STRIPE_SECRET
     DJSTRIPE_WEBHOOK_SECRET = os.environ.get("RTD_DJSTRIPE_WEBHOOK_SECRET")
 
+    @property
+    def SOCIALACCOUNT_PROVIDERS(self):
+        """Allow settings social account settigs from the host system."""
+        providers = self._SOCIALACCOUNT_PROVIDERS
+        for provider in providers.keys():
+            try:
+                for setting in ["client_id", "secret", "key"]:
```
Comment (stsewd) [inline readthedocs/settings/docker_compose.py:217]: We can probably just ignore key, none of the providers make use of it

**Comment by agjohnson** [comment]
Updated with some docs, dropped the unused `key`, and updated `common` submodule

File: docs/dev/settings.rst [hunk 2]
```diff
@@ -130,5 +130,43 @@ ELASTICSEARCH_DSL_AUTOSYNC
...
+providers using the following environment variables:
+
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_SECRET
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GITLAB_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GITLAB_SECRET
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_OAUTH2_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_OAUTH2_SECRET
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_SECRET
```
Comment (humitos) [inline docs/dev/settings.rst:161]: These two can be deleted. They are deprecated and we are not using them anymore, even in production.

File: docs/dev/settings.rst [hunk 3]
```diff
@@ -130,5 +130,43 @@ ELASTICSEARCH_DSL_AUTOSYNC
...
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GOOGLE_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GOOGLE_SECRET
+
+Stripe secrets
+~~~~~~~~~~~~~~
+
+The following secrets are required to use `djstripe` and our Stripe integration.
```
Comment (stsewd) [inline docs/dev/settings.rst:168]:
```suggestion
The following secrets are required to use ``djstripe`` and our Stripe integration.
```

**Review by stsewd — APPROVED** [review]

File: docs/dev/settings.rst (hunk 2 above)
Comment (agjohnson) [reply docs/dev/settings.rst:161]: I am going to drop the documentation for these, but `settings.SOCIALACCOUNT_PROVIDERS` will still load these env vars if they exist as it's just looping over the providers defined. I think this is fine for now though, and removing the underlying provider settings will end this.

Comment (stsewd) [reply docs/dev/settings.rst]: We have an open issue for removing that one from our settings https://github.com/readthedocs/readthedocs.org/issues/11162.

**Pull request merged**